File: LittleProjects/Javascript/ReactFluxMySQLCrud/core/service/entityService.js
import pool from '../configuration/dataSource'
import repository from '../repository/entityRepository'

class EntityService {
    constructor() {
    }
}

EntityService.prototype.findOne = (res) => (id) => {
    pool.getConnection(function (err, connection) {
        if (err) {
            console.log(err);
            return;
        }

        try {
            repository.findOne(connection, res)(id);
        } catch (e) {
            console.log(e);
        } finally {
            connection.release();
        }
    });
};

EntityService.prototype.findAll = (res) => () => {
    pool.getConnection(function (err, connection) {
        if (err) {
            console.log(err);
            return;
        }

        try {
            repository.findAll(connection, res)();
        } catch (e) {
            console.log(e);
        } finally {
            connection.release();
        }
    });
};

EntityService.prototype.save = (res) => (entity) => {
    pool.getConnection(function (err, connection) {
        if (err) {
            console.log(err);
            return;
        }

        try {
            repository.save(connection, res)(entity);
        } catch (e) {
            console.log(e);
        } finally {
            connection.release();
        }
    });

};

EntityService.prototype.deleteById = (res) => (id) => {
    pool.getConnection(function (err, connection) {
        if (err) {
            console.log(err);
            return;
        }

        try {
            repository.deleteById(connection, res)(id);
        } catch (e) {
            console.log(e);
        } finally {
            connection.release();
        }
    });

};

EntityService.prototype.update = (res) => (entity) => {
    pool.getConnection(function (err, connection) {
        if (err) {
            console.log(err);
            return;
        }

        try {
            repository.update(connection, res)(entity);
        } catch (e) {
            console.log(e);
        } finally {
            connection.release();
        }
    });

};

const service = new EntityService();

export default service;

File: LittleProjects/Scala/AkkaQuillSQLCrud/src/main/scala/akka/quill/sql/crud/service/EntityService.scala
package akka.quill.sql.crud.service

import akka.quill.sql.crud.model.Entity
import akka.quill.sql.crud.configuration.DataSource.ctx
import akka.quill.sql.crud.repository.EntityRepository

import scala.util.Try

class EntityService extends Service[Entity, Long] {
  val repository = EntityRepository()

  override def findOne(id: Long): Either[Throwable, Option[Entity]] = Try {
    ctx.transaction {
      repository.findOne(id).value.unsafeRunSync()
    }
  }.toEither

  override def findAll(): Either[Throwable, Seq[Entity]] = Try {
    ctx.transaction {
      repository.findAll().unsafeRunSync()
    }
  }.toEither

  override def save(a: Entity): Either[Throwable, Entity] = Try {
    ctx.transaction {
      repository.save(a).unsafeRunSync()
    }
  }.toEither

  override def update(a: Entity): Either[Throwable, Entity] = Try {
    ctx.transaction {
      repository.update(a).unsafeRunSync()
    }
  }.toEither

  override def deleteOne(id: Long): Either[Throwable, Long] = Try {
    ctx.transaction {
      repository.deleteOne(id).unsafeRunSync()
    }
  }.toEither

  override def deleteAll(): Either[Throwable, Long] = Try {
    ctx.transaction {
      repository.deleteAll().unsafeRunSync()
    }
  }.toEither
}

object EntityService {
  def apply(): EntityService = new EntityService()
}
